src/backend/BackendInterface.cpp

```diff
@@ -0,0 +1,361 @@
+#include <ripple/protocol/Indexes.h>
+#include <ripple/protocol/STLedgerEntry.h>
+#include <backend/BackendInterface.h>
+namespace Backend {
+bool
+BackendInterface::finishWrites(uint32_t ledgerSequence) const
+{
+    indexer_.finish(ledgerSequence, *this);
+    auto commitRes = doFinishWrites();
+    if (commitRes)
+    {
+        if (isFirst_)
+            indexer_.doKeysRepairAsync(*this, ledgerSequence);
+        if (indexer_.isKeyFlagLedger(ledgerSequence))
+            indexer_.writeKeyFlagLedgerAsync(ledgerSequence, *this);
+        isFirst_ = false;
+    }
+    else
+    {
+        // if commitRes is false, we are relinquishing control of ETL. We
+        // reset isFirst_ to true so that way if we later regain control of
+        // ETL, we trigger the index repair
+        isFirst_ = true;
+    }
+    return commitRes;
+}
+bool
+BackendInterface::isLedgerIndexed(std::uint32_t ledgerSequence) const
+{
+    auto keyIndex = getKeyIndexOfSeq(ledgerSequence);
+    if (keyIndex)
+    {
+        auto page = doFetchLedgerPage({}, ledgerSequence, 1);
+        return !page.warning.has_value();
+    }
+    return false;
+}
+void
+BackendInterface::writeLedgerObject(
+    std::string&& key,
+    uint32_t seq,
+    std::string&& blob) const
+{
+    assert(key.size() == sizeof(rippled::uint256));
```

Comment:
I expect you're mostly compiling in release mode. This line doesn't compile (`ripple::uint256` not `rippled::uint256`)